src/python/pants/reporting/report.py

```diff
@@ -130,7 +130,9 @@ def _notify(self):
     # than the current one, if work is happening in parallel.
     # Assumes self._lock is held by the caller.
     for workunit in self._workunits.values():
-      for label, output in workunit.outputs().items():
+      # N.B. Wrap .items() call with list() due to issues with workunit.outputs()
+      # changing its size during iteration.
+      for label, output in list(workunit.outputs().items()):
```

Comment:
This was not needed in Py2 because `.items()` always returns a new list in Py2, whereas in Py3 it returns a lazy dict items `view`.

Reply:
Could this comment briefly mention or link to the issues causing `workunit.outputs()` to change its size during iteration?